nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ControllerResource.java

```diff
@@ -494,22 +495,26 @@ public Response createFlowRegistryClient(
         preprocessObsoleteRequest(requestFlowRegistryClientEntity);
 
         if (requestFlowRegistryClientEntity == null || requestFlowRegistryClientEntity.getComponent() == null) {
-            throw new IllegalArgumentException("Flow registry client details must be specified.");
+            throw new IllegalArgumentException("Flow Registry client details must be specified.");
         }
 
         if (requestFlowRegistryClientEntity.getRevision() == null
                 || (requestFlowRegistryClientEntity.getRevision().getVersion() == null
                 || requestFlowRegistryClientEntity.getRevision().getVersion() != 0)) {
-            throw new IllegalArgumentException("A revision of 0 must be specified when creating a new Registry.");
+            throw new IllegalArgumentException("A revision of 0 must be specified when creating a new Flow Registry.");
         }
 
         final FlowRegistryClientDTO requestRegistryClient = requestFlowRegistryClientEntity.getComponent();
         if (requestRegistryClient.getId() != null) {
-            throw new IllegalArgumentException("Registry ID cannot be specified.");
+            throw new IllegalArgumentException("Flow Registry ID cannot be specified.");
         }
 
         if (StringUtils.isBlank(requestRegistryClient.getName())) {
-            throw new IllegalArgumentException("Registry name must be specified.");
+            throw new IllegalArgumentException("Flow Registry name must be specified.");
+        }
+
+        if (serviceFacade.getRegistryClients().stream().map(rce -> rce.getComponent().getName()).collect(Collectors.toSet()).contains(requestRegistryClient.getName())) {
```

Comment:
```suggestion
        if (serviceFacade.getRegistryClients().stream().anyMatch(rce -> requestRegistryClient.getName().equals(rce.getComponent().getName()))) {
```

I think use of the `anyMatch` stream function is more obvious than collecting everything into a `Set` and checking for `contains`